nautobot/extras/tests/test_views.py

```diff
@@ -908,3 +922,370 @@ def setUpTestData(cls):
             "http_method": "POST",
             "http_content_type": "application/json",
         }
+
+
+@override_settings(JOBS_ROOT=THIS_DIRECTORY)
+class TestJobMixin(SimpleTestCase):
+    class TestJob(Job):
+        pass
+
+    @staticmethod
+    def get_test_job_class(self, class_path):
+        if class_path.startswith("local/test_view"):
+            return TestJobMixin.TestJob
+        raise Http404
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # Monkey-patch the viewsets' _get_job methods to return our test class above
+        cls.original_method = JobView._get_job
+        JobView._get_job = TestJobMixin.get_test_job_class
+        ScheduledJobView._get_job = TestJobMixin.get_test_job_class
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        # Undo monkey-patch
+        JobView._get_job = cls.original_method
+        ScheduledJobView._get_job = cls.original_method
+
+
+class ScheduledJobTestCase(
+    TestJobMixin,
+    ViewTestCases.GetObjectViewTestCase,
+    ViewTestCases.ListObjectsViewTestCase,
+    ViewTestCases.DeleteObjectViewTestCase,
+):
+    model = ScheduledJob
+
+    @classmethod
+    def setUpTestData(cls):
+        user = User.objects.create(username="user1", is_active=True)
+        ScheduledJob.objects.create(
+            name="test1",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="local/test_views/TestJob",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            start_time=datetime.now(),
+        )
+        ScheduledJob.objects.create(
+            name="test2",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="local/test_views/TestJob",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            start_time=datetime.now(),
+        )
+        ScheduledJob.objects.create(
+            name="test3",
+            task="nautobot.extras.jobs.scheduled_job_handler",
+            job_class="local/test_views/TestJob",
+            interval=JobExecutionType.TYPE_IMMEDIATELY,
+            user=user,
+            start_time=datetime.now(),
+        )
+        # this should not appear, since it’s not enabled
```

Comment:
I think the intermittent failure in CI is because `test_list_objects_with_constrained_permission()` just grabs the last two objects in `self._get_queryset()`, so unless we have a defined ordering for this queryset, it's going to get two more-or-less at random, which might or might not include this object.

I think you can fix it by *only* creating this object inside the `test_only_enabled_is_listed` test case, instead of class-wide here in `setUpTestData`.

Reply:
Tried it out, thanks for the tip!

Reply:
That seems to have worked, yay!